src/NuGet.Clients/NuGet.PackageManagement.UI/ViewModels/PackageDetailsTabViewModel.cs

```diff
@@ -0,0 +1,122 @@
+        }
+
+        public async Task SetCurrentFilterAsync(ItemFilter filter)
+        {
+            if (_readmeTabEnabled)
+            {
+                await ReadmePreviewViewModel.SetCurrentFilterAsync(filter);
+            }
+        }
+
+        private void ReadMePreviewViewModel_PropertyChanged(object sender, PropertyChangedEventArgs e)
+        {
+            if (ReadmePreviewViewModel is not null
+                && e.PropertyName == nameof(ReadmePreviewViewModel.ReadmeMarkdown))
+            {
+                Tabs[0].Visible = !string.IsNullOrWhiteSpace(ReadmePreviewViewModel.ReadmeMarkdown);
+                if (!Tabs[0].Visible)
+                {
+                    SelectedTab = Tabs[1];
+                }
+            }
+        }
+
+        private void DetailControlModel_PropertyChanged(object sender, PropertyChangedEventArgs e)
+        {
+            ThreadHelper.JoinableTaskFactory.Run(async () =>
+            {
+                if (_readmeTabEnabled)
+                {
+                    await ReadmePreviewViewModel.SetPackageMetadataAsync(DetailControlModel.PackageMetadata, CancellationToken.None);
+                }
+            });
+        }
+
+        public async Task SetPackageMetadataAsync(DetailControlModel detailControlModel, CancellationToken cancellationToken)
+        {
+            if (_readmeTabEnabled)
+            {
+                await ReadmePreviewViewModel.SetPackageMetadataAsync(detailControlModel.PackageMetadata, cancellationToken);
+                RaisePropertyChanged(nameof(ReadmePreviewViewModel));
+            }
+
+            DetailControlModel.PropertyChanged -= DetailControlModel_PropertyChanged;
+            Tabs[1] = detailControlModel;
+            DetailControlModel = detailControlModel;
+            DetailControlModel.PropertyChanged += DetailControlModel_PropertyChanged;
+            RaisePropertyChanged(nameof(DetailControlModel));
+        }
+
+        private bool _disposed = false;
+        public void Dispose()
+        {
+            if (_disposed)
+            {
+                return;
+            }
+            _disposed = true;
+            DetailControlModel.PropertyChanged -= DetailControlModel_PropertyChanged;
+        }
+
+        internal void SelectTab(PackageMetadataTab selectedPackageMetadataTab)
+        {
+            switch (selectedPackageMetadataTab)
+            {
+                case PackageMetadataTab.Readme:
+                    SelectedTab = Tabs[0];
+                    break;
+                case PackageMetadataTab.PackageDetails:
+                    SelectedTab = Tabs[1];
+                    break;
+            }
+        }
+
+        public ReadmePreviewViewModel ReadmePreviewViewModel { get; private set; }
+
+        public DetailControlModel DetailControlModel { get; private set; }
+
+        public ObservableCollection<TabViewModelBase> Tabs { get; private set; }
```

Comment:
I don't think this collection of Tabs changes, so I'd just use a simpler List that's readonly and create/initialize the list in the `Initialize`.

```suggestion
        public IReadOnlyList<TabViewModelBase> Tabs { get; private set; }
```

Reply:
I do set it in the code though I don't know how often DetailControlModel changes, From the breakpoints doesn't seem like it does.